pygpu/gpuarray.pyx

```diff
@@ -1415,10 +1412,7 @@ cdef GpuArray pygpu_index(GpuArray a, const ssize_t *starts,
                           const ssize_t *stops, const ssize_t *steps):
     cdef GpuArray res
     res = new_GpuArray(type(a), a.context, a.base)
-    try:
-        array_index(res, a, starts, stops, steps)
-    except ValueError, e:
-        raise IndexError, "index out of bounds"
+    array_index(res, a, starts, stops, steps)
```

Comment:
I think this is the reason for the newly-failing test in https://github.com/Theano/Theano/pull/6026#issuecomment-307196300
I'm not sure what is happening, but presumably what used to be an IndexError was catched somewhere, and now it is a Value Error.